src/scene/vcModel.cpp

```diff
@@ -285,6 +286,92 @@ void vcModel::HandleImGui(vcState *pProgramState, size_t * /*pItemID*/)
+    ImGui::Indent();
+    for (uint32_t i = 0; i < m_pointCloudHeader.attributes.count; ++i)
+    {
+      ImGui::BulletText("%s", m_pointCloudHeader.attributes.pDescriptors[i].name);
+
+      if (pProgramState->settings.presentation.showDiagnosticInfo)
+      {
+        ImGui::Indent();
+        switch (m_pointCloudHeader.attributes.pDescriptors[i].typeInfo)
+        {
+        case vdkAttributeTypeInfo_uint8:
+          vcStringFormat(buffer, udLengthOf(buffer), vcString::Get("attributeType"), vcString::Get("attributeTypeUint8"));
+          break;
+        case vdkAttributeTypeInfo_uint16:
+          vcStringFormat(buffer, udLengthOf(buffer), vcString::Get("attributeType"), vcString::Get("attributeTypeUint16"));
+          break;
+        case vdkAttributeTypeInfo_uint32:
+          vcStringFormat(buffer, udLengthOf(buffer), vcString::Get("attributeType"), vcString::Get("attributeTypeUint32"));
+          break;
+        case vdkAttributeTypeInfo_uint64:
+          vcStringFormat(buffer, udLengthOf(buffer), vcString::Get("attributeType"), vcString::Get("attributeTypeUint64"));
+          break;
+        case vdkAttributeTypeInfo_int8:
+          vcStringFormat(buffer, udLengthOf(buffer), vcString::Get("attributeType"), vcString::Get("attributeTypeInt8"));
+          break;
+        case vdkAttributeTypeInfo_int16:
+          vcStringFormat(buffer, udLengthOf(buffer), vcString::Get("attributeType"), vcString::Get("attributeTypeInt16"));
+          break;
+        case vdkAttributeTypeInfo_int32:
+          vcStringFormat(buffer, udLengthOf(buffer), vcString::Get("attributeType"), vcString::Get("attributeTypeInt32"));
+          break;
+        case vdkAttributeTypeInfo_int64:
+          vcStringFormat(buffer, udLengthOf(buffer), vcString::Get("attributeType"), vcString::Get("attributeTypeInt64"));
+          break;
+        case vdkAttributeTypeInfo_float32:
+          vcStringFormat(buffer, udLengthOf(buffer), vcString::Get("attributeType"), vcString::Get("attributeTypeFloat32"));
+          break;
+        case vdkAttributeTypeInfo_float64:
+          vcStringFormat(buffer, udLengthOf(buffer), vcString::Get("attributeType"), vcString::Get("attributeTypeFloat64"));
+          break;
+        case vdkAttributeTypeInfo_color32:
+          vcStringFormat(buffer, udLengthOf(buffer), vcString::Get("attributeType"), vcString::Get("attributeTypeColour"));
+          break;
+        case vdkAttributeTypeInfo_normal32:
+          vcStringFormat(buffer, udLengthOf(buffer), vcString::Get("attributeType"), vcString::Get("attributeTypeNormal"));
+          break;
+        case vdkAttributeTypeInfo_vec3f32:
+          vcStringFormat(buffer, udLengthOf(buffer), vcString::Get("attributeType"), vcString::Get("attributeTypeVec3F32"));
+          break;
+        default:
+          vcStringFormat(buffer, udLengthOf(buffer), vcString::Get("attributeType"), vcString::Get("attributeTypeUnknown"));
+          break;
+        }
```

Comment:
I'd prefer a map where these enum indices map directly to the fetched strings

Reply:
That would leave us in a state where we need to alloc and cleanup the map and ultimately would be the same amount of code and complexity.

This is untidy but its explicit in what it's doing.

Reply:
in this instance I mean map as in an array where indices map to the vcString entry, no need to dynamically alloc or clean up and presumbly more efficient in that this cascading switch structure is replace by a calculation then jump

Reply:
Just to clarify, are you suggesting something like:
```c++
const char *pAttrStringMap[vdkAttributeTypeInfo_Count] = {
  vcString::Get("attributeTypeUint8"),
  ...
};
vcStringFormat(buffer, udLengthOf(buffer), vcString:Get("attributeType"), pAttrStringMap[m_pointCloudHeader.attributes.pDescriptors[i].typeInfo]);
```
I don't think those VDK values are sequential from 0, so an array wouldn't really work?

Reply:
Sorry, I misunderstood; @samsinsane is right, these are not-sequential. The values are encoded data about the type.

The values (encoded) are:
```
    vdkAttributeTypeInfo_uint8 = 1,
    vdkAttributeTypeInfo_uint16 = 2,
    vdkAttributeTypeInfo_uint32 = 4,
    vdkAttributeTypeInfo_uint64 = 8,
    vdkAttributeTypeInfo_int8 = 65537,
    vdkAttributeTypeInfo_int16 = 65538,
    vdkAttributeTypeInfo_int32 = 65540,
    vdkAttributeTypeInfo_int64 = 65544,
    vdkAttributeTypeInfo_float32 = 196612,
    vdkAttributeTypeInfo_float64 = 196616,
    vdkAttributeTypeInfo_color32 = 262148,
    vdkAttributeTypeInfo_normal32 = 524292,
    vdkAttributeTypeInfo_vec3f32 = 197388
```

The `vdkAttributes` header has more information on these types and how to decode them